hugegraph-core/src/main/java/com/baidu/hugegraph/util/Reflection.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2017 HugeGraph Authors
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.util;
+
+import com.baidu.hugegraph.HugeException;
+import com.baidu.hugegraph.exception.NotSupportException;
+import org.slf4j.Logger;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+public class Reflection {
+
+    private static final Logger LOG = Log.logger(Reflection.class);
+
+    private static final Class reflectionClazz;
+    private static final Method registerFieldsToFilterMethod;
+    private static final Method registerMethodsToFilterMethod;
+
+    public static final String JDK_INTERNAL_REFLECT_REFLECTION =
+                               "jdk.internal.reflect.Reflection";
+    public static final String SUN_REFLECT_REFLECTION =
+                               "sun.reflect.Reflection";
+
+    static {
+        Method registerFieldsToFilterMethodTemp = null;
+        Method registerMethodsToFilterMethodTemp = null;
+        Class reflectionClazzTemp = null;
+        try {
+            reflectionClazzTemp = Class.forName(
+                                  JDK_INTERNAL_REFLECT_REFLECTION);
+        } catch (ClassNotFoundException e) {
+            try {
+                reflectionClazzTemp = Class.forName(SUN_REFLECT_REFLECTION);
+            } catch (ClassNotFoundException ex) {
+                LOG.error("Can't find Reflection class", ex);
```

Comment:
should throw HugeException here?

Reply:
logger error may be more best, because java 17 don't this API

Reply:
we maybe should use the java9 module to define runtime API and compile API visible, and remove this class